gaphor/storage/storage.py

```diff
@@ -587,7 +587,8 @@ def upgrade_simple_properties_to_value_specifications(
                     )
                     if type is not None and isinstance(value, str):
                         upperValue = element_factory.create(type)
-                        upperValue.value = "*" if value == "*" else int(value)
+                        # Overwrite any string value that is not '*'
+                        upperValue.value = "*" if isinstance(value, str) else int(value)
```

Comment:
This will not yield the expected result since value is always a `str` (see line 558) Instead it could be something like:

```python
try:
    upperValue.value = int(value)
except ValueError:
    upperValue.value = "*"
```